### PR TITLE
Initial commit.

### PR DESCRIPTION
**Pull request by ababicheva**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+# Android Studio Navigation editor temp files
+.navigation/
+
+# Android Studio captures folder
+captures/
+
+# Intellij
+*.iml
+.idea/*
```
Comment (yakivmospan) [inline .gitignore:38]: Somehow, idea folders were still committed.

Comment (yakivmospan) [reply .gitignore:38]: Should we add general codestyle to this repo ?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,2 @@
 # android-ktx
-Android Kotlin extensions library.
+Android Kotlin extensions library made by Temyco Android developers.
```
Comment (yakivmospan) [inline README.md]: `made by Temyco `, just "Temy" ?

File: app/build/generated/source/buildConfig/debug/co/temy/android/ktx/BuildConfig.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Automatically generated file. DO NOT MODIFY
+ */
+package co.temy.android.ktx;
```
Comment (yakivmospan) [inline app/build/generated/source/buildConfig/debug/co/temy/android/ktx/BuildConfig.java]: Why auto-generated file is included?

File: app/src/main/res/values/styles.xml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<resources>
```
Comment (yakivmospan) [inline app/src/main/res/values/styles.xml]: Do we need styles for this kind of library?

File: app/src/main/res/values/colors.xml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
```
Comment (yakivmospan) [inline app/src/main/res/values/colors.xml]: Same for colors and launch icons

File: app/src/main/java/co/temy/android/ktx/context/ContextExtensions.kt [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    } else {
+        IconCompat.createWithResource(this, vectorIconId)
+    }
+}
+
+/**
+ * Navigates to the current Google user's account
+ * subscriptions in Google Play app.
+ */
+fun Activity.openGooglePlaySubscriptions() {
```
Comment (yakivmospan) [inline app/src/main/java/co/temy/android/ktx/context/ContextExtensions.kt]: Should we move all Activity extensions in separate files?

File: app/src/main/java/co/temy/android/ktx/crypto/CipherExtensions.kt [hunk 7]
```diff
@@ -0,0 +1,98 @@
+package co.temy.android.ktx.crypto
```
Comment (yakivmospan) [inline app/src/main/java/co/temy/android/ktx/crypto/CipherExtensions.kt]: Would be nice to have some sample documentation, maybe with another pull request in future we can provide them

File: app/src/main/java/co/temy/android/ktx/view/ViewExtensions.kt [hunk 8]
```diff
@@ -0,0 +1,558 @@
...
+    setPadding(paddingLeft, paddingTop, paddingRight, bottomPadding)
+}
+
+fun Context.inflate(resource: Int, root: ViewGroup? = null, attachToRoot: Boolean = false): View {
+    return LayoutInflater.from(this).inflate(resource, root, attachToRoot)
+}
+
+fun Context.getColorCompat(@ColorRes color: Int) = ContextCompat.getColor(this, color)
+
+fun Context.getDrawableCompat(@DrawableRes drawable: Int) = ContextCompat.getDrawable(this, drawable)
```
Comment (yakivmospan) [inline app/src/main/java/co/temy/android/ktx/view/ViewExtensions.kt]: Part of Contex extensions ?

File: app/src/main/java/co/temy/android/ktx/view/ViewExtensions.kt [hunk 9]
```diff
@@ -0,0 +1,558 @@
...
+ */
+fun Int.toDp(): Int {
+    val metrics = Resources.getSystem().displayMetrics
+    return this / (metrics.densityDpi / DisplayMetrics.DENSITY_DEFAULT)
+}
+
+/**
+ * Converts value in device-independent pixels into value in pixels
+ */
+fun Int.toPx(): Int {
```
Comment (yakivmospan) [inline app/src/main/java/co/temy/android/ktx/view/ViewExtensions.kt]: Also, a question, if this should be a part of View extensions, or maybe Resource, Contex ? ..

File: app/src/main/java/co/temy/android/ktx/view/ViewExtensions.kt [hunk 10]
```diff
@@ -0,0 +1,558 @@
...
+    }
+
+    return spannable
+}
+
+/**
+ * Returns [Spannable] where all <em> (e.g. italic) tags are replaced
+ * with <strong> (e.g. bold)
+ *
+ * Use it to show a search result for /pages response.
```
Comment (yakivmospan) [inline app/src/main/java/co/temy/android/ktx/view/ViewExtensions.kt]: We should remove our project related comments from docs

File: app/src/main/java/co/temy/android/ktx/view/ViewExtensions.kt [hunk 11]
```diff
@@ -0,0 +1,558 @@
...
+            spannable.getSpanEnd(it),
+            Spannable.SPAN_EXCLUSIVE_EXCLUSIVE
+        )
+        spannable.removeSpan(it)
+    }
+}
+
+fun ActionMenuView.inflateMenu(menuId: Int) = SupportMenuInflater(context).inflate(menuId, menu)
+
+fun Intent.getNullableIntExtra(name: String, default: Int? = null): Int? {
```
Comment (yakivmospan) [inline app/src/main/java/co/temy/android/ktx/view/ViewExtensions.kt]: Not a part of view extentions

File: app/src/main/java/co/temy/android/ktx/view/ViewExtensions.kt [hunk 12]
```diff
@@ -0,0 +1,558 @@
...
+    val extra = getIntExtra(name, -1)
+    return if (extra == -1) default else extra
+}
+
+fun Context.currentLanguage() = ConfigurationCompat.getLocales(resources.configuration)[0].language
```
Comment (yakivmospan) [inline app/src/main/java/co/temy/android/ktx/view/ViewExtensions.kt]: Same here

**Review by yakivmospan — COMMENTED** [review]
In general:
- We need to update gitignore
- We need to remove style resources, this library shall have no such items
- We need to think how to separate extensions better
- We need to update documentation for extensions

File: app/src/main/java/co/temy/android/ktx/view/ViewExtensions.kt (hunk 11 above)
Comment (ababicheva) [reply app/src/main/java/co/temy/android/ktx/view/ViewExtensions.kt]: strange extension at all. removed it

File: app/src/main/java/co/temy/android/ktx/view/ViewExtensions.kt (hunk 8 above)
Comment (ababicheva) [reply app/src/main/java/co/temy/android/ktx/view/ViewExtensions.kt]: we must discuss it on android fridays.

File: app/src/main/AndroidManifest.xml [hunk 13]
```diff
@@ -0,0 +1,11 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="co.temy.android.ktx">
+
+    <application
+        android:allowBackup="true"
+        android:icon="@mipmap/ic_launcher"
```
Comment (yakivmospan) [inline app/src/main/AndroidManifest.xml]: Same as resources and styles, do we need launcher icon for lib project?

File: app/src/main/java/co/temy/android/ktx/view/ViewExtensions.kt (hunk 8 above)
Comment (yakivmospan) [reply app/src/main/java/co/temy/android/ktx/view/ViewExtensions.kt]: Don't remember if we discussed it already, but let's follow the original caller for an extension? `ContextCompat` => `ContextExtentions`?

**Review by yakivmospan — APPROVED** [review]

**Pull request merged**